python/LAMBDA_INSIDE_VPC/LAMBDA_INSIDE_VPC.py

```diff
@@ -0,0 +1,422 @@
+
+Scenarios:
+  Scenario: 1
+    Given: No Lambda functions are present
+    Then: Return NOT_APPLICABLE
+
+  Scenario: 2
+    Given: Lambda function is present in the VPC i.e. VPC enabled:true
+    And: Rule Parameter subnetId is empty
+    Then: Return COMPLIANT
+
+  Scenario: 3
+    Given: Lambda function is not present in the VPC i.e. VPC enabled:false
+    Then: Return NON_COMPLIANT
+
+  Scenario: 4
+    Given: Lambda function is present in the VPC i.e. VPC enabled:true
+    And: Rule Parameter list doesn't match with the subnetId
+    Then: Return NON_COMPLIANT
+
+  Scenario: 5
+    Given: Lambda function is present in the VPC i.e. VPC enabled:true
+    And: Rule Parameter list matches with the subnetId
+    Then: Return COMPLIANT
+"""
+
+import json
+import sys
+import datetime
+import boto3
+import botocore
+
+try:
+    import liblogging
+except ImportError:
+    pass
+
+##############
+# Parameters #
+##############
+
+# Define the default resource to report to Config Rules
+DEFAULT_RESOURCE_TYPE = 'AWS::Lambda::Function'
+
+# Set to True to get the lambda to assume the Role attached on the Config Service (useful for cross-account).
+ASSUME_ROLE_MODE = False
+
+# Other parameters (no change needed)
+CONFIG_ROLE_TIMEOUT_SECONDS = 900
+
+#############
+# Main Code #
+#############
+
+def evaluate_compliance(event, configuration_item, valid_rule_parameters):
+    # if "vpcConfig" is not present, then lambda function is outside the VPC
+    if 'vpcConfig' not in configuration_item['configuration']:
+        return build_evaluation_from_config_item(configuration_item, 'NON_COMPLIANT', annotation='This Lambda Function is not in VPC')
+
+    # if "vpcConfig" exists but no subnet is present, this scenario is possilbe if the lambda function was previously part of a VPC.
+    if not configuration_item['configuration']['vpcConfig']['subnetIds']:
+        return build_evaluation_from_config_item(configuration_item, 'NON_COMPLIANT', annotation='This Lambda Function is not in VPC')
+
+    # if no input parameter provided then return COMPLIANT
+    if not valid_rule_parameters:
+        return build_evaluation_from_config_item(configuration_item, 'COMPLIANT')
+
+    # compare the subnets of lambda with the input parameter using set operation
+    if set(configuration_item['configuration']['vpcConfig']['subnetIds']) - set(valid_rule_parameters):
+        return build_evaluation_from_config_item(configuration_item, 'NON_COMPLIANT', annotation='This Lambda Function is not associated with the subnets specified in the "subnetId" input parameter')
+
+    return build_evaluation_from_config_item(configuration_item, 'COMPLIANT')
+
+
+def evaluate_parameters(rule_parameters):
+    # if paramter is given, check whether its subnet Id or not, else ignore it as it is an optional paramter
+    if "subnetId" not in rule_parameters:
+        return {}
+
+    # if paramter is given but value is empty then ignore it.
+    if not rule_parameters['subnetId']:

```

Comment:
This cannot happen. Config will not pass an empty parameter.

Reply:
I tried giving below value for the `InputParameters` with the empty value for the subnetId and empty value to the InputParameters,

```
    "InputParameters": "{\"subnetId\": \"\"}"
    "InputParameters": "{}"
```

In both cases RDK got deployed successfully and on the console it was showing Null as value for the SubnetId in the first and entire parameter as Null in second case.

When i tried giving empty value on the console (GUI) it ignore if its empty and does not create any parameter at all.

Looks like RDK is accepting the empty value, in that case should I keep the check or remove it ?